File: design-patterns/go/decorator/data_export/excel_data_export_decorator.go
// excel_data_export_decorator.go

package main

import "fmt"

type ExcelDataExportDecorator struct {
	*DataExportDecorator
}

func NewExcelDataExportDecorator(dataExporter DataExport) (excelDataExportDecorator *ExcelDataExportDecorator) {
	excelDataExportDecorator = &ExcelDataExportDecorator{}
	excelDataExportDecorator.DataExportDecorator = NewDataExportDecorator(dataExporter)
	return
}

func (excelDataExportDecorator *ExcelDataExportDecorator) ProcessData() {
	excelDataExportDecorator.DataExportDecorator.ProcessData()
	excelDataExportDecorator.processExcel()
}

func (excelDataExportDecorator *ExcelDataExportDecorator) processExcel() {
	fmt.Println("Processed data to Excel")
}

File: design-patterns/php/decorator/dataexport/ExcelDataExportDecorator.php
<?php
// ExcelDataExportDecorator.php

namespace BigBoxCode\DesignPattern\Decorator\DataExport;

class ExcelDataExportDecorator extends DataExportDecorator {
    public function __construct(DataExport $dataExporter) {
        parent::__construct($dataExporter);
    }

    public function processData(): void {
        $this->dataExporter->processData();
        $this->processExcel();
    }

    public function processExcel() {
        echo "Processed data to Excel\n";
    }
}